pyfar/io.py

```diff
@@ -187,3 +192,88 @@ def _sofa_pos(pos_type):
     else:
         raise ValueError("Position:Type {pos_type} is not supported.")
     return domain, convention, unit
+
+
+def read(filename):
+    """
+    Read any compatible pyfar format from disk.
+
+    Parameters
+    ----------
+    filename : string or open file handle.
+        Input file must be haiopy compatible.
+
+    Returns
+    -------
+    collection: dictionary
+        containing PyFar types like
+        { 'name1': 'obj1', 'name2': 'obj2' ... }.
+    """
+    # Check for .far file extension
+    if filename.split('.')[-1] != 'far':
+        warnings.warn("Extending filename by .far.")
+        filename += '.far'
+
+    collection = {}
+    with open(filename, 'rb') as f:
+        zip_buffer = io.BytesIO()
+        zip_buffer.write(f.read())
+        with zipfile.ZipFile(zip_buffer) as zip_file:
+            zip_paths = zip_file.namelist()
+            obj_names = set([path.split('/')[0] for path in zip_paths])
+            for name in obj_names:
+                json_str = zip_file.read(name + '/json').decode('UTF-8')
+                obj_type, obj_dict = json.loads(json_str)
+                obj_dict = codec._decode(obj_dict, zip_file)
+                ObjType = codec._str_to_type(obj_type)
+                obj = ObjType._decode(obj_dict)
+                if not codec._is_pyfar_type(obj):
+                    raise TypeError(
+                        f'Objects of type {type(obj)}'
+                        'cannot be read from disk.')
+                collection[name] = obj
+
+    return collection
+
+
+def write(filename, compress=False, **objs):
+    """
+    Write any compatible pyfar format to disk.
+
+    Parameters
+    ----------
+    filename : string or open file handle.
+        Input file must be pyfar compatible.
+    compress : bool
+        Default is false (uncompressed).
+        If false zipfile.ZIP_STORED mode is used,
+        if True, zipfile.ZIP_DEFLATED mode is used.
```

Comment:
I am not sure if this is an improvement since the original comment provides more information and I expect people who use the library to have a general understanding of what is 'compressed'. But I'll follow your suggestion for encapsulation reasons.